relay-event-schema/src/protocol/span.rs

```diff
@@ -177,6 +165,85 @@ impl Getter for Span {
     }
 }
 
+#[derive(Clone, Debug, Default, PartialEq, Empty, FromValue, IntoValue, ProcessValue)]
+#[cfg_attr(feature = "jsonschema", derive(JsonSchema))]
+pub struct SpanData {
+    /// The source code file name that identifies the code unit as uniquely as possible.
+    #[metastructure(field = "code.filepath", pii = "maybe")]
+    pub code_filepath: Annotated<Value>,
+    /// The line number in `code.filepath` best representing the operation.
+    #[metastructure(field = "code.lineno", pii = "maybe")]
+    pub code_lineno: Annotated<Value>,
+    /// The method or function name, or equivalent.
+    ///
+    /// Usually rightmost part of the code unit's name.
+    #[metastructure(field = "code.function", pii = "maybe")]
+    pub code_function: Annotated<Value>,
+    /// The "namespace" within which `code.function` is defined.
+    ///
+    /// Usually the qualified class or module name, such that
+    /// `code.namespace + some separator + code.function`
+    /// form a unique identifier for the code unit.
+    #[metastructure(field = "code.namespace", pii = "maybe")]
+    pub code_namespace: Annotated<Value>,
```

Comment:
We could give these fields stricter types, but

1. We don't want parsing of `span.data` to suddenly fail because of stricter types,
2. Holding a `Value` allows us to return a `&Annotated<Value>` from `.get()`, to give `SpanData`  an `Object`-like interface.